src/components/utils/OffchainSigner/api/requests.ts

```diff
@@ -1,16 +1,86 @@
-import { isFunction } from '@polkadot/util'
+import { AxiosError } from 'axios'
 import { OffchainSignerEndpoint, sendHttpRequest, SendHttpRequestProps } from './utils'
 
-export const requestMessage = async (
-  myAddress: string,
-  onFailedCallback?: (err: Error) => void,
-) => {
-  if (!myAddress) {
-    throw new Error('No account id provided')
+type EmailSignUpProps = {
+  email: string
+  password: string
+  accountAddress: string
+  signedProof: string
+  proof: string
+  hcaptchaResponse: string
+}
+
+type EmailSignInProps = {
+  email: string
+  password: string
+  hcaptchaResponse: string
+}
+
+type AddressSignInProps = {
+  signedProof: string
+  proof: string
+  hcaptchaResponse: string
+}
+
+type ConfirmEmailProps = {
+  code: string
+  accessToken: string
+}
+
+type SubmitSignedCallDataProps = {
+  data: string
+  jwt: string
+}
+
+export type JwtPayload = {
+  accountAddress: string
+  accountAddressVerified: boolean
+  email: string
+  emailVerified: boolean
+  iat: number
+  exp: number
+}
+interface Error {
+  message: string[]
+  statusCode: number
+}
+
+type ErrorMessageProps = {
+  error: unknown
+  showError?: boolean
+}
+
+export function onErrorHandler(
+  error: unknown,
+  callback: (errorMessage: string) => void,
+  showError?: boolean,
+) {
+  const errorMessage = getErrorMessage({ error, showError })
+  if (errorMessage) {
+    callback(errorMessage as string)
   }
+}
 
+export function getErrorMessage({ error, showError = false }: ErrorMessageProps) {
+  const err = error as AxiosError<Error>
+  if (!err.response?.data) return err.message
+  const { message, statusCode } = err.response?.data
+
+  // Handle specific error message from resend confirmation endpoint
+  if (
+    statusCode === 400 &&
+    typeof message === 'string' &&
+    message === 'Wait before sending another confirmation.' &&
```

Comment:
It is message from back, the check looks strange for me

Reply:
True, I've already applied a UX solution so we don't need this check anymore.

Removed the check in 1406bb3